### PR TITLE
Python 3 fixes

### PR DESCRIPTION
**Pull request by egbertbouman**

### CODE REVIEW

File: ipv8/attestation/trustchain/block.py [hunk 1]
```diff
@@ -445,8 +445,8 @@ def __iter__(self):
         for key, value in self.__dict__.items():
             if key == 'key' or key == 'serializer' or key == 'crypto' or key == '_transaction':
                 continue
-            if isinstance(value, string_types) and key != "insert_time" and key != "type":
-                yield key, hexlify(value)
+            if isinstance(value, binary_type) and key != "insert_time" and key != "type":
```
Comment (qstokkink) [inline ipv8/attestation/trustchain/block.py]: Wait, this doesn't seem correct. If it's binary type then the matching with the unicode strings will never pass.

Comment (egbertbouman) [reply ipv8/attestation/trustchain/block.py]: Which matching?

Comment (qstokkink) [reply ipv8/attestation/trustchain/block.py]: `key != "insert_time" and key != "type"`

Comment (qstokkink) [reply ipv8/attestation/trustchain/block.py]: Under Python 3 you cannot compare bytes and str:

```python
> print(b"insert_time" == "insert_time")
False
```

Comment (egbertbouman) [reply ipv8/attestation/trustchain/block.py]: It's matching the key but converting the value. If the value is of type bytes, then it encodes to hex.

Comment (qstokkink) [reply ipv8/attestation/trustchain/block.py]: Oops, my bad.

**Pull request merged**